components/script/bluetooth_blacklist.rs

```diff
@@ -0,0 +1,114 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+use regex::Regex;
+use std::cell::RefCell;
+use std::collections::HashMap;
+use std::fs::File;
+use std::io::{BufRead, BufReader};
+use std::string::String;
+
+const EXCLUDE_READS: &'static str = "exclude-reads";
+const EXCLUDE_WRITES: &'static str = "exclude-writes";
+const FILE_PATH: &'static str = "./resources/gatt_blacklist.txt";
+const VALID_UUID_REGEX: &'static str = "^[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}";
+
+thread_local!(pub static BLUETOOTH_BLACKLIST: RefCell<BluetoothBlacklist> =
+              RefCell::new(BluetoothBlacklist(parse_blacklist())));
+
+macro_rules! return_if_blacklisted {
+    ($uuid:expr, $exclude_fn:ident) => {
+        if BLUETOOTH_BLACKLIST.with(|blist| {
+            blist.borrow().$exclude_fn($uuid.as_ref())
+        }) {
+            return Err(Security);
+        }
+    }
+}
+
+pub struct BluetoothBlacklist(HashMap<String, ExcludeType>);
+
+#[derive(Eq, PartialEq)]
+pub enum ExcludeType {
+    Exclude, // Read and Write
+    ExcludeReads,
+    ExcludeWrites,
+}
+
+impl BluetoothBlacklist {
+
+    // https://webbluetoothcg.github.io/web-bluetooth/#blacklisted
+    pub fn is_excluded(&self, uuid: &str) -> bool {
```

Comment:
` is_blacklisted `, `is_blacklisted_for_reads ` and ` is_blacklisted_for_writes` ?